moonstone/analysis/diversity/alpha.py

```diff
@@ -0,0 +1,88 @@
+from abc import ABC
+
+from typing import Union, Optional
+
+import pandas as pd
+import re
+import skbio
+
+from moonstone.core.module_base import BaseModule, BaseDF
+from moonstone.plot.graphs.histogram import Histogram
+from moonstone.utils.plot import (
+    add_default_titles_to_plotting_options
+)
+
+
+class BaseAnalysis(BaseModule, BaseDF, ABC):
+    """
+    ????
+    """
+    def __init__(self, dataframe: Union[pd.Series, pd.DataFrame]):
+        super().__init__(dataframe)
+
+
+class AlphaDiversity(BaseModule, BaseDF, ABC):
+
+    def __init__(self, dataframe: Union[pd.Series, pd.DataFrame]):
+        super().__init__(dataframe)
+        self.index_name = " ".join(re.findall('[A-Z][^A-Z]*', self.__class__.__name__)).lower()
```

Comment:
I am not sure what you want to do here. Generate the name dynamically based on the class name?

Reply:
Yes, to have it inside the title and xlabel

Reply:
Because it's the only thing changing between the visualize of the two classes. Maybe I could have use super() and declare different title and xlabel depending on the class ? But I don't really know how

Reply:
Works for now I think!